cpp/dolfinx/common/sort.h

```diff
@@ -23,144 +27,81 @@ namespace dolfinx
 /// @tparam T Integral type
 /// @tparam BITS The number of bits to sort at a time.
 /// @param[in, out] array The array to sort.
-template <typename T, int BITS = 8>
-void radix_sort(std::span<T> array)
+struct __radix_sort
```

Comment:
Is it necessary to pack the method into a struct?

Reply:
This follows the style of the STL regarding the handling of functors and enables to use the `radix_sort` in any other STL algorithm as an inpute function.

The new design with theses structs follows the realization of lambdas, and removes the necessity of handling function pointers as template types. This reduces the amount of necessary type dispatching drastically if I remember correctly.

What we could do is move the implementation of the `radix_sort` into an anonymous namespace to separate it from its wrapping for external use.

Reply:
Thanks for educating me!